mssql-ballerina/Client.bal

```diff
@@ -0,0 +1,212 @@
+        if (self.clientActive) {
+            return nativeCall(self, sqlQuery, rowTypes);
+        } else {
+            return error sql:ApplicationError("MsSQL Client is already closed, hence further operations are not allowed");
+        }
+    }
+
+    # Close the SQL client.
+    #
+    # + return - Possible error during closing the client
+    public isolated function close() returns sql:Error? {
+        self.clientActive = false;
+        return close(self);
+    }
+
+}
+
+# Client Configuration record for connection initialization
+#
+# + host - Hostname of the mssql server to be connected
+# + instance - Instance name of the server to be connected as mssql can have installations of multiple versions 
+#              under a single server
+# + port - Port number of the mssql server to be connected
+# + database - Name of the database
+# + user - Username for the database connection
+# + password - Password for the database connection
+# + options - MsSQL database specific options
+# + connectionPool - The `sql:ConnectionPool` object to be used within 
+#         the jdbc client. If there is no connectionPool provided, 
+#         the global connection pool will be used
+type ClientConfiguration record {|
+    string host;
+    string? instance;
+    int port;
+    string? user;
+    string? password;
+    string? database;
+    Options? options;
+    sql:ConnectionPool?  connectionPool;
+|};
+
+# MsSQL database options.
+#
+# + secureSocket - SSL Configuration to be used
+# + socketTimeout - The number of milliseconds to wait before a timeout is occurred 
+#                   on a socket read or accept. The default value is 0, which means 
+#                   infinite timeout
+# + queryTimeout - The number of seconds to wait before a timeout has occurred on a 
+#                  query. The default value is -1, which means infinite timeout.
+#                  Setting this to 0 also implies to wait indefinitely
+# + loginTimeout - The number of seconds the driver should wait before timing out a 
+#                 failed connection. A zero value indicates that the timeout is the 
+#                 default system timeout, which is specified as 15 seconds by default. 
+#                 A non-zero value is the number of seconds the driver should wait 
+#                 before timing out a failed connection
+
+public type Options record {|
+    SecureSocket secureSocket = {};
+    decimal socketTimeout?;
+    decimal queryTimeout?;
+    decimal loginTimeout?;
+|};
+
+# SSL configuration to be used when connecting to the MsSQL server
+#
+# + encrypt - encryption for all the data sent between the client and the server if the server has a certificate installed
+# + trustServerCertificate - If "true", the SQL Server SSL certificate is automatically trusted when the communication layer is encrypted using TLS
+
+public type SecureSocket record {|
+    boolean encrypt?;
+    boolean trustServerCertificate?;
```

Comment:
How can we pass secure socket cert file, Keystore, etc to establish secure socket?

Reply:
I couldn't find a default file for sql server to test SSL connection so as for now we can  use the sql server self-signed certificate by setting the properties `encrypt=true` and `trustServerCertificate=true`.